### PR TITLE
127466 alerts gui

### PR DESCRIPTION
**Pull request by nilsgstrabo**
Manage alerting for environment and pipeline jobs

Waiting for feedback from EDS on design

### CODE REVIEW

File: src/components/alerting/edit-alerting.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import { UpdateAlertingConfigModel } from '../../models/alerting';
+import PropTypes from 'prop-types';
+import { TextField } from '@equinor/eds-core-react';
+import * as React from 'react';
```
Comment (Pespiri) [inline src/components/alerting/edit-alerting.js:4]: Avoid importing entire components where possible
```suggestion
import { useEffect, useState } from 'react';
```

**Review by Pespiri — CHANGES_REQUESTED** [review]

**Comment by Pespiri** [comment]
Also see if you can use a more modern pattern for the state (reducers and such)

File: src/components/alerting/alerting-overview.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  config: PropTypes.shape(AlertingConfigModel).isRequired,
+};
+
+const AlertingEnabled = ({ config }) => {
+  return (
+    <>
+      {config.ready ? (
+        <>
+          <AlertingReady config={config} />
+        </>
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply src/components/alerting/alerting-overview.js]: Fixed

File: src/components/alerting/edit-alerting.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+const buildSlackReceiverNamesFromConfig = (config) => {
+  if (!config && !config.receivers) {
+    return [];
+  }
+
+  return Object.entries(config.receivers)
+    .filter((e) => e[1].slackConfig)
+    .map((e) => e[0]);
+};
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply src/components/alerting/edit-alerting.js]: Fixed

File: src/components/alerting/edit-alerting.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+              placeholder="Enter a Slack webhook URL where alerts should be sent"
+              onChange={(ev) =>
+                slackUrlChangeCallback(receiver, ev.target.value)
+              }
+            />
+          </React.Fragment>
+        ))}
+    </>
+  );
+};
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply src/components/alerting/edit-alerting.js]: Fixed

File: src/components/alerting/edit-alerting.js (hunk 1 above)
Comment (nilsgstrabo) [reply src/components/alerting/edit-alerting.js]: fixed

**Review by Pespiri — DISMISSED** [review]

**Review by Pespiri — APPROVED** [review]

**Pull request merged**